src/utils/date.ts

```diff
@@ -164,25 +194,41 @@ export function parseNumber(str: string, localeData: DateLocaleData): number {
 }
 
 /**
- * Parse numeric units for day, month, and year from a localized string
+ * Parse day, month, and year from a localized string
  * month starts at 0 (can pass to date constructor)
+ * can return values as number or string
  *
  * @param str
  * @param localeData
+ * @param returnType
  */
-export function parseDateString(str: string, localeData: DateLocaleData): { day: number; month: number; year: number } {
+export function parseDateString(
+  str: string,
+  localeData: DateLocaleData,
+  returnType: "number" | "string" = "number"
```

Comment:
Can we leave it up to the component to stringify the numbers? I'm assuming this is to avoid parsing from string to format again, but I think this is negligible. We can revisit if we notice perf issues.

Reply:
The `string` returnType is not just for convenience.  It allows for checking if a user entered years with leading zeros like `0001`, `0010`, `0100`, which is the centerpiece of the original issue, which is to prevent the calendar from updating on years less than 1000.

I could create a separate util function that returns `day`, `month` and `year` as strings, or a simpler function like `isFourDigitYear` that literally just checks if a date value contains at least a 4 digit year, but both those seem unnecessary and redundant because I still would have to handle `unitOrder` and `separator` from the locale data.

Reply:
Per our conversation, I like your suggestions to add a method that parses a string into the specific parts. We could keep `parseDateString`'s signature the same and internally it could use the date splitting method and it would just need to convert them to numbers. My main reason was to keep the return type simple and consistent and not have to think about switching things around. It's only used in the date picker for now, so I guess it's not a big deal to have a flag for the return type.

Reply:
Ok, I created a new function `datePartsFromLocalizedString` that moves the logic for capturing the day, month and year values as strings from an incoming localized string so that we can properly determine if a 4-digit year is typed and not have to overload the return type on the parseDateString function.  Let me know if that looks good :)